### DISCUSSION TITLE
Object of class DateTime could not be converted to string

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46333] [Symfony Book: The Fast Track]
Hello everyOne
I have a history entity that has as attribute datetime , value , and register_id 
the datetime and register_id are composite primary key 
when i want to persist an object history i keep get **Object of class DateTime could not be converted to string**

**Comment by stof** [comment]
This has nothing to do with Symfony (Doctrine is a separate project).

and this is a documented limitation of Doctrine: https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/tutorials/composite-primary-keys.html#primitive-types-only

**Reply by Elmatbaisaad** [reply]
The probleme come from persist the history object here is what i did
![Capture d’écran 2022-05-12 120450](https://user-images.githubusercontent.com/101296742/168046558-5c22b515-fdce-4366-a275-2fcd6838239e.png)
![Capture d’écran 2022-05-12 120417](https://user-images.githubusercontent.com/101296742/168046574-a8505888-e1c9-46d7-a4ac-3280ef58dff6.png)

**Reply by stof** [reply]
the problem comes from using a datetime field as part of your compound id, while Doctrine says this supports only scalar fields (or relations).
